components/tor/tor_launcher_factory.cc

```diff
@@ -260,40 +274,61 @@ void TorLauncherFactory::GotSOCKSListeners(
     observer.OnTorNewProxyURI(tor_proxy_uri);
 }
 
-void TorLauncherFactory::OnTorClosed() {
-  DCHECK_CURRENTLY_ON(BrowserThread::UI);
+void TorLauncherFactory::OnTorControlClosed(bool was_running) {
+  SEQUENCE_CHECKER(sequence_checker_);
   VLOG(2) << "TOR CONTROL: Closed!";
+  // If we're still running, try watching again to start over.
+  // XXX Rate limit in case of flapping?
```

Comment:
is it todo?

Reply:
Looks like it is TODO, I'm not the original author, just moving the comment around for refactoring.

Reply:
I changed it to TODO with original author's handle in aeed04b506e574973c5df12c9781dde63e187342